joomla/joomla-cms
# [4.0] Plugin to add "CAPTCHA - hCaptcha" functionality

Pull Request for Issue https://github.com/joomla/joomla-cms/issues/28613
~~**Note:** due to a bug with ALL Captcha's in Joomla 4, you will need to manually install PR https://github.com/joomla/joomla-cms/pull/28785 in case it has NOT yet been merged into Joomla 4.~~ The mentioned Captcha issue has been solved & merged into Joomla. You can test this PR against Joomla 4.0-dev

### Summary of Changes
This plugin adds "hCaptcha" as alternative CAPTCHA method in Joomla forms.

### Testing Instructions
In Joomla 4 back-end:
- Install the "CAPTCHA - hCaptcha" plugin (via System > Install > Discover)
- Enable + **configure** the **hCaptcha** plugin: 
System > Plugins > open the hCaptcha plugin
  - Add a Site Key + Secret Key
  - Enable the plugin
- **Set** hCaptcha as **default CAPTCHA** plugin
System > Global Configuration > [Site] tab > Default Captcha: select "**CAPTCHA - hCaptcha**"
- **Create** a **contact form**
Components > Contacts > Contacts > New
  - Add "Test" Name + Email
- **Create menu item** for **contact form**
Menus > Main Menu > New > 
  - Add menu title "Contact form"
  - Select "Menu Item Type": Contacts > Single Contact
  - Select "Contact": the "Test" contact that you've just created

On the front-end: Click on the "Contact form" menu item that you've just created.
At the bottom you should have the CAPTCHA with the hCaptcha form.
